### PR TITLE
feat: adding script to unseal Vault

### PR DESCRIPTION
**Pull request by iwalmsley**
Have tested this locally and with multiple keys on the test cluster

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM node:8.16.1-alpine
```
Comment (gary-lloyd) [inline Dockerfile]: Given this is a new repo we should probably be using Node 12 as the current version. We should really be looking to move all other services to this too and will need to do so before Jan 2020 when Node 8 maintenance finishes - https://nodejs.org/en/about/releases/

File: src/unseal.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+function unsealVault(key) {
+  return axios.post(`${VAULT_ADDR}/v1/sys/unseal`, { key });
+}
+
+sealStatus()
+  .then((sealed) => {
+    if (sealed) {
+      console.log('Vault is sealed, unsealing...');
+      KEYS_ARR.forEach(unsealVault);
```
Comment (gary-lloyd) [inline src/unseal.js]: I don't think this works as intended in the failure case as you don't return the result of all of the two underlying promises. You'll need to use `Promise.all`. Alternatively, if you have updated to Node 12 you'll be able to use async/await to make it imperative code.

File: src/unseal.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const axios = require('axios');
+
+const { VAULT_KEYS, VAULT_ADDR } = process.env;
```
Comment (gary-lloyd) [inline src/unseal.js]: Might it be worth using the `convict` library to handle environment variables? You'll get better errors for missing configuration that way and also the ability to set default for VAULT_ADDR.

**Review by gary-lloyd — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (iwalmsley) [reply Dockerfile]: Have updated that

File: src/unseal.js (hunk 2 above)
Comment (iwalmsley) [reply src/unseal.js]: I have refactored this, but only based on googling etc, does it make sense now?

File: src/unseal.js (hunk 3 above)
Comment (iwalmsley) [reply src/unseal.js]: Switched to that

File: src/unseal.js (hunk 2 above)
Comment (gary-lloyd) [reply src/unseal.js]: I think this still doesn't do what we want for the reasons discussed. Here is the refactored snippet. You definitely need to test it works though!

```
async function sealStatus() {
  const response = await axios.get(`${vaultAddress}/v1/sys/seal-status`);
  return res.data;
}

function unsealVault(key) {
  return axios.post(`${vaultAddress}/v1/sys/unseal`, { key });
}

async function unsealChecker() {
  try {
    const sealed = await sealStatus();

    if (sealed) {
      console.log('Vault is currently sealed');
      const unsealPromises = KEYS_ARR.map(unsealVault);
      await Promise.all(unsealPromises);
    } else {
      console.log('Vault is currently unsealed');
    }
  } catch(err) {
    console.log(`Error: ${err}.`);
  }
}

unsealChecker();
```

**Review by gary-lloyd — CHANGES_REQUESTED** [review]

**Comment by iwalmsley** [comment]
That seems to be working plus a few tweaks

**Review by gary-lloyd — APPROVED** [review]
Looks good. I guess we now need to get DockerHub building the container before we can add to the manifests to deploy

**Pull request merged**